### PR TITLE
feat: refactor publish/offline fe

### PR DESCRIPTION
**Pull request by liuxiran**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] New feature provided

- Related issues

FE of https://github.com/apache/apisix-dashboard/issues/837

depends on:

- APISIX release with https://github.com/apache/apisix/pull/2943

- BE: https://github.com/apache/apisix-dashboard/pull/1049 (should be mergered first)

### CODE REVIEW

File: web/src/pages/Route/List.tsx [hunk 1]
```diff
@@ -84,17 +116,38 @@ const Page: React.FC = () => {
             >
               {formatMessage({ id: 'component.global.edit' })}
             </Button>
+            <Button
+              type="primary"
+              onClick={() => {
+                handlePublishOffline(record.id, 1)
```
Comment (LiteSun) [inline web/src/pages/Route/List.tsx:122]: IMHO, use `Enum` will more readable.🤔

Comment (liuxiran) [reply web/src/pages/Route/List.tsx]: fixed, thanks

File: web/src/pages/Route/List.tsx [hunk 2]
```diff
@@ -23,12 +23,31 @@ import { PlusOutlined } from '@ant-design/icons';
...
+    notification.success({
+      message: msgTip,
+    });
+
+    ref.current?.reload();
+  };
+
+  const handlePublishOffline = (rid: string, status: RouteModule.RouteStatus) => {
+    updateRouteStatus(rid, status).then(() => {
+      const actionName = status? formatMessage({ id: 'page.route.publish' }) : formatMessage({ id: 'page.route.offline' })
```
[the remark this answers is not in the document]
Comment (liuxiran) [reply web/src/pages/Route/List.tsx]: fixed, thanks

**Comment by liuxiran** [comment]
trying to analyze the reason for the e2e failure🧐

**Comment by LiteSun** [comment]
> trying to analyze the reason for the e2e failure🧐

Here is the reason why front-end e2e failed: 

field `status` not accept by the current branch manage-api, will sovled after https://github.com/apache/apisix-dashboard/pull/1049 merged.

image

cc @liuxiran

**Comment by liuxiran** [comment]
> > trying to analyze the reason for the e2e failure🧐
> 
> Here is the reason why front-end e2e failed:
> 
> field `status` not accept by the current branch manage-api, will sovled after #1049 merged.
> 
> image
> 
> cc @liuxiran

Got it, thanks a lot @LiteSun , when  #1049 merged, I will rerun the CI ^_^

**Review by juzhiyuan — APPROVED** [review]

**Review by LiteSun — APPROVED** [review]

**Pull request merged**